### PR TITLE
CI Processor: simply return None (instead of raising error) when a me…

### LINKED ISSUE #67: CI support needs more work
Please note that CI messages under eng.ci.* are used by multiple teams, we need to adjust few fields there.

Also the processor seems sensitive to invalid messages:

```
{"detail":"string indices must be integers","error":"internal_error","traceback":"Traceback (most recent call last):\n  File \"/usr/lib/python2.7/site-packages/flask/app.py\", line 1982, in wsgi_app\n    response = self.full_dispatch_request()\n  File \"/usr/lib/python2.7/site-packages/flask/app.py\", line 1614, in full_dispatch_request\n    rv = self.handle_user_exception(e)\n  File \"/usr/lib/python2.7/site-packages/flask/app.py\", line 1517, in handle_user_exception\n    reraise(exc_type, exc_value, tb)\n  File \"/usr/lib/python2.7/site-packages/flask/app.py\", line 1612, in full_dispatch_request\n    rv = self.dispatch_request()\n  File \"/usr/lib/python2.7/site-packages/flask/app.py\", line 1598, in dispatch_request\n    return self.view_functions[rule.endpoint](**req.view_args)\n  File \"/usr/lib/python2.7/site-packages/datagrepper/app.py\", line 421, in raw\n    message = message_card(msg, size)\n  File \"/usr/lib/python2.7/site-packages/datagrepper/util.py\", line 117, in message_card\n    subtitle = fedmsg.meta.msg2subtitle(msg, legacy=False, **config)\n  File \"/usr/lib/python2.7/site-packages/fedmsg/meta/__init__.py\", line 152, in __wrapper\n    return f(msg, **config)\n  File \"/usr/lib/python2.7/site-packages/fedmsg/meta/__init__.py\", line 171, in __wrapper\n    return f(msg, processor=processor, **config)\n  File \"/usr/lib/python2.7/site-packages/fedmsg/meta/__init__.py\", line 246, in msg2subtitle\n    return processor.subtitle(msg, **config)\n  File \"/usr/lib/python2.7/site-packages/fedmsg_meta_umb/ci.py\", line 38, in subtitle\n    return ('Test job for ' + msg['msg']['artifact']['nvr'] + ' ' +\nTypeError: string indices must be integers\n"}
```

### PR DESCRIPTION
**Pull request by YukinChan**
…ssage do not contain required field

This should fixes #67 .

### CODE REVIEW

File: fedmsg_meta_umb/ci.py [hunk 1]
```diff
@@ -39,16 +39,22 @@ def subtitle(self, msg, **config):
                         msg['topic'].split('.')[-1])
             except KeyError:
```
Comment (waynesun09) [inline fedmsg_meta_umb/ci.py]: As both return None, could caught in one line with:

except (KeyError, TypeError):

File: fedmsg_meta_umb/ci.py [hunk 2]
```diff
@@ -39,16 +39,22 @@ def subtitle(self, msg, **config):
...
             return set()
+        except TypeError:
+            return None
 
     def link(self, msg, **config):
         try:
             return msg['msg']['run']['url']
         except KeyError:
             return None
+        except TypeError:
```
Comment (waynesun09) [inline fedmsg_meta_umb/ci.py]: same here

File: fedmsg_meta_umb/ci.py [hunk 3]
```diff
@@ -39,16 +39,22 @@ def subtitle(self, msg, **config):
...
+                return None
         return None
 
     def packages(self, msg, **config):
         try:
             return set([msg['headers']['component']])
         except KeyError:
             return set()
+        except TypeError:
+            return None
```
Comment (mikebonnet) [inline fedmsg_meta_umb/ci.py]: This should return an empty `set()`, as it does above.

File: fedmsg_meta_umb/ci.py (hunk 1 above)
Comment (mikebonnet) [reply fedmsg_meta_umb/ci.py]: +1

**Pull request merged**